src/rules/preferReadonlyRule.ts

```diff
@@ -0,0 +1,189 @@
+
+            default:
+                ts.forEachChild(node, this.visitNode);
+        }
+    }
+
+    private handleArrowFunctionOrFunctionExpression(node: ts.ArrowFunction | ts.FunctionExpression) {
+        if (this.scope === undefined) {
+            return;
+        }
+
+        this.scope.enterNonConstructorScope();
+        ts.forEachChild(node, this.visitNode);
+        this.scope.exitNonConstructorScope();
+    }
+
+    private handleClassDeclarationOrExpression(node: ts.ClassDeclaration | ts.ClassExpression) {
+        const parentScope = this.scope;
+        const childScope = this.scope = new ClassScope(node, this.typeChecker);
+
+        ts.forEachChild(node, this.visitNode);
+
+        this.finalizeScope(childScope);
+        this.scope = parentScope;
+    }
+
+    private handleConstructor(node: ts.ConstructorDeclaration) {
+        this.scope!.enterConstructor();
+
+        for (const parameter of node.parameters) {
+            if (utils.isModfierFlagSet(parameter, ts.ModifierFlags.Private)) {
+                this.scope!.addDeclaredVariable(parameter);
+            }
+        }
+
+        ts.forEachChild(node, this.visitNode);
+        this.scope!.exitConstructor();
+    }
+
+    private handlePostfixOrPrefixUnaryExpression(node: ts.PostfixUnaryExpression | ts.PrefixUnaryExpression) {
+        if (this.scope === undefined) {
+            return;
+        }
+
+        this.scope.addVariableModification(node.operand as ts.PropertyAccessExpression);
+    }
+
+    private handlePropertyDeclaration(node: ts.PropertyDeclaration) {
+        this.scope!.addDeclaredVariable(node);
+        ts.forEachChild(node, this.visitNode);
+    }
+
+    private handlePropertyAccessExpression(node: ts.PropertyAccessExpression, parent: ts.Node) {
+        switch (parent.kind) {
+            case ts.SyntaxKind.BinaryExpression:
+                this.handleParentBinaryExpression(node, parent as ts.BinaryExpression);
+                break;
+
+            case ts.SyntaxKind.DeleteExpression:
+                this.handleDeleteExpression(node);
+        }
+
+        ts.forEachChild(node, this.visitNode);
+    }
+
+    private handleParentBinaryExpression(node: ts.PropertyAccessExpression, parent: ts.BinaryExpression) {
+        if (this.scope !== undefined && parent.left === node && utils.isAssignmentKind(parent.operatorToken.kind)) {
+            this.scope.addVariableModification(node);
+        }
+    }
+
+    private handleDeleteExpression(node: ts.PropertyAccessExpression) {
+        if (this.scope !== undefined) {
+            this.scope.addVariableModification(node);
+        }
+    }
+
+    private finalizeScope(scope: ClassScope) {
+        for (const violatingNode of scope.finalizeUnmodifiedPrivateNonReadonlys()) {
+            this.complainOnNode(violatingNode);
+        }
+    }
+
+    private complainOnNode(node: ParameterOrPropertyDeclaration) {
+        const fix = Lint.Replacement.appendText(node.name.getStart(), "readonly ");
```

Comment:
the new modifier is automatically placed next to the others. you don't have to worry about comments, line breaks, etc.